start-site/src/main/java/io/spring/start/site/extension/springcloud/DefaultSpringCloudProjectsMetadataProvider.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2012-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.spring.start.site.extension.springcloud;
+
+import java.util.Collections;
+import java.util.Map;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * The default {@link SpringCloudProjectsMetadataProvider} implementation that caches the
+ * {@link SpringCloudProjectsMetadata}.
+ *
+ * @author Olga Maciaszek-Sharma
+ */
+class DefaultSpringCloudProjectsMetadataProvider
+		implements SpringCloudProjectsMetadataProvider {
+
+	private static final Log LOG = LogFactory
+			.getLog(DefaultSpringCloudProjectsMetadataProvider.class);
+
+	private final RestTemplate restTemplate;
+
+	private final String infoHost;
+
+	DefaultSpringCloudProjectsMetadataProvider(RestTemplate restTemplate,
```

Comment:
Let's use `WebClient` from now on. I'll add an issue to migrate our current use.